src/main/java/com/atlassian/db/replica/api/DualConnection.java

```diff
@@ -375,25 +375,45 @@ public PreparedStatement prepareStatement(String sql, String[] columnNames) thro
     @Override
     public Clob createClob() throws SQLException {
         checkClosed();
-        throw new ReadReplicaUnsupportedOperationException();
+        if(compatibleWithPreviousVersion)
+            throw new ReadReplicaUnsupportedOperationException();
+        else
+            return connectionProvider
+                .getWriteConnection(new RouteDecisionBuilder(Reason.RW_API_CALL))
```

Comment:
Why write-connection?

Reply:
I added commit message to explain reasoning behind this

Reply:
It's not pushed yet